platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java

```diff
@@ -811,55 +812,52 @@ private VirtualFileEntry getVirtualFile(String workspace, String path, boolean f
     }
 
     private ImportResponse configureProject(ImportProject importProject, FolderEntry baseProjectFolder, String workspace, long creationDate)
-            throws IOException, ForbiddenException, ConflictException, NotFoundException,
-                   ServerException {
+            throws IOException, ForbiddenException, ConflictException, NotFoundException, ServerException, BadRequestException {
+        NewProject newProject = importProject.getProject();
+        if (newProject == null) {
+            throw new BadRequestException("Project that importing can not be null");
+        }
         ImportResponse importResponse = DtoFactory.getInstance().createDto(ImportResponse.class);
         ProjectTypeRegistry projectTypeRegistry = projectManager.getProjectTypeRegistry();
         Project project;
         ProjectDescriptor projectDescriptor;
-        ProjectConfig projectConfig = null;
+        ProjectConfig projectConfig;
         String visibility = null;
-        NewProject newProject = importProject.getProject();
+
         //try convert folder to project with giving config
         try {
-            if (newProject != null) {
-                visibility = newProject.getVisibility();
-                RunnersDescriptor descriptor = newProject.getRunners();
-                if (descriptor != null) {
-                    replaceDefaultRunnerPath(descriptor);
-                }
-                projectConfig = DtoConverter.fromDto2(newProject, projectTypeRegistry);
-            }
+            visibility = newProject.getVisibility();
+            RunnersDescriptor descriptor = newProject.getRunners();
```

Comment:
if it be null then this method throw npe here https://github.com/codenvy/che-core/blob/master/platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java#L888